src/main/kotlin/org/jetbrains/research/ml/coding/assistant/unification/DatasetUnification.kt

```diff
@@ -0,0 +1,59 @@
+package org.jetbrains.research.ml.coding.assistant.unification
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.Project
+import com.intellij.psi.codeStyle.CodeStyleManager
+import org.jetbrains.research.ml.coding.assistant.dataset.model.DatasetRecord
+import org.jetbrains.research.ml.coding.assistant.dataset.model.DynamicSolution
+import org.jetbrains.research.ml.coding.assistant.solutionSpace.utils.psiCreator.PsiCreator
+import org.jetbrains.research.ml.coding.assistant.unification.model.DatasetPartialSolution
+import org.jetbrains.research.ml.coding.assistant.unification.model.DynamicIntermediateSolution
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.logging.Logger
+
+@Service
+interface DatasetUnification {
+    /**
+     * Unifies the given solution dynamic
+     * @param dynamicSolution raw dataset dynamic solution
+     * @return unified solution using ast transformation
+     */
+    fun transform(dynamicSolution: DynamicSolution): DynamicIntermediateSolution
+}
+
+class DatasetUnificationImpl(private val project: Project) : DatasetUnification {
+    private val logger = Logger.getLogger(javaClass.name)
+    private val fileFactory: PsiCreator = project.service()
+    private val codeStyleManager = CodeStyleManager.getInstance(project)
+
+    override fun transform(dynamicSolution: DynamicSolution): DynamicIntermediateSolution {
+        val datasetRecords = dynamicSolution.records
+
+        val counter = AtomicInteger(0)
+        return datasetRecords
+            .map {
+                val counterValue = counter.incrementAndGet()
+                logger.info { "Start unify(id=${it.id}) $counterValue/${datasetRecords.size}" }
+                unifyRecord(it)
+            }
+            .toList()
+    }
+
+    private fun unifyRecord(datasetRecord: DatasetRecord): DatasetPartialSolution {
+        val psiFile = fileFactory.initFileToPsi(datasetRecord.fragment)
+        WriteCommandAction.runWriteCommandAction(project) { // reformat the expected file
+            codeStyleManager.reformat(psiFile)
+        }
```

Comment:
I think we should create a function for it. Also we can use this function in the tests